### PR TITLE
chore: bridge-api fetchBridgeQuotes util

### PR DESCRIPTION
**Pull request by micaelae**
## **Description**
Changes
- Implement the `fetchBridgeQuotes` util to be used for making requests to the bridge-api `getQuotes` endpoint
- Copy in bridge-api types
- Create test data

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/28027?quickstart=1)

## **Related issues**

Fixes: https://consensyssoftware.atlassian.net/browse/MMS-1448

## **Manual testing steps**

N/A

### **Before**

N/A

### **After**

N/A

## **Pre-merge author checklist**

- [X] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [X] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X]  I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by ejwessel — DISMISSED** [review]

File: test/data/bridge/mock-quotes-erc20-erc20.ts
[the remark this answers is not in the document]
Comment (micaelae) [reply test/data/bridge/mock-quotes-erc20-erc20.ts:1]: Updated this!

**Review by ejwessel — DISMISSED** [review]

File: ui/pages/bridge/bridge.util.ts [hunk 1]
```diff
@@ -159,3 +154,105 @@ export async function fetchBridgeTokens(
...
+    .map(([k, v]) => `${k}=${v}`)
+    .join('&')}`;
+  const quotes = await fetchWithCache({
+    url,
+    fetchOptions: { method: 'GET', headers: CLIENT_ID_HEADER },
+    cacheOptions: { cacheRefreshTime: 0 },
+    functionName: 'fetchBridgeQuotes',
+  });
+
+  const filteredQuotes = quotes.filter((quote: QuoteResponse) =>
```
Comment (infiniteflower) [inline ui/pages/bridge/bridge.util.ts:172]: Would it be cleaner to use a nested validator for those object validation functions?

File: ui/pages/bridge/types.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+      (field) =>
+        field in partialRequest &&
+        typeof partialRequest[field as keyof typeof partialRequest] ===
+          'number' &&
+        partialRequest[field as keyof typeof partialRequest] !== undefined &&
+        !isNaN(Number(partialRequest[field as keyof typeof partialRequest])) &&
+        partialRequest[field as keyof typeof partialRequest] !== null,
+    )
+  );
+};
```
Comment (infiniteflower) [inline ui/pages/bridge/types.ts:54]: Should this be in a utils file?

Comment (micaelae) [reply ui/pages/bridge/types.ts:54]: good call! moving this

File: ui/pages/bridge/bridge.util.ts [hunk 3]
```diff
@@ -159,3 +154,105 @@ export async function fetchBridgeTokens(
...
   return transformedTokens;
 }
+
+// Returns a list of bridge tx quotes
+export async function fetchBridgeQuotes(
+  request: QuoteRequest,
+): Promise<QuoteResponse[]> {
+  const url = `${BRIDGE_API_BASE_URL}/getQuote?${Object.entries(request)
+    .map(([k, v]) => `${k}=${v}`)
+    .join('&')}`;
```
[the remark this answers is not in the document]
Comment (micaelae) [reply ui/pages/bridge/bridge.util.ts:164]: It requires more code bc each non-string `request` entry needs to be stringified, but I think it's better to set the query params explicitly anyway. Will do this

**Review by infiniteflower — APPROVED** [review]

**Review by ejwessel — APPROVED** [review]

**Pull request merged**